pgsanity/pgsanity.py

```diff
@@ -27,8 +48,12 @@ def check_file(filename=None, show_filename=False):
     else:
         with sys.stdin as filelike:
             sql_string = sys.stdin.read()
-
-    success, msg = check_string(sql_string)
+    # check for BOM-table and discard if present
+    nose = sql_string[0:len(BOM_UTF8)]
+    bom_present = check_for_bom(nose)
+    sql_string = sql_string[len(nose):] if bom_present else sql_string
+    success, msg = check_string(sql_string.decode("utf-8"))
+    # ^ The above call to decode() is safe for both ASCII and UTF-8 data.
```

Comment:
This is a bit nit-picky, but I think I'd rather see all of this BOM stuff move in to check_for_bom(), which probably means that check_for_bom() would become remove_bom_if_exists().  It would take the full sql string and return the full sql string (with the BOM removed if it exists).  I think that encapsulates the BOM handling a bit more.